src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -776,10 +796,206 @@ def __try_add_error(error_list, detail):
         }
+
+    def __set_error_detail(self, error_code, message):
+        formatted_message = self.__ensure_error_message_restriction_compliance(message)
+        return {
+            "code": str(error_code),
+            "message": str(formatted_message)
+        }
+    # endregion
+
+    # region - Patch Truncation
+    def get_truncated_patches(self):
+        return self.__truncated_patches
+
+    def __write_truncated_status_file(self, status_file_payload):
+        """ Truncate the substatus summary patch list when complete status file size is more than 126kb """
+        complete_status_byte_size = self.__get_byte_size(status_file_payload)
+        truncated_status_file = status_file_payload
+
+        if complete_status_byte_size > Constants.MAX_STATUS_FILE_SIZE_IN_BYTES:
+
+            self.composite_logger.log("Begin Truncation")
+            assessment_name = self.__assessment_substatus_json['name']
+            assessment_patch = self.__assessment_packages
+            errors_detail_list = []
+
+            # Truncated assessment patch when operation is not installation
+            if self.execution_config.operation != Constants.INSTALLATION.lower() and assessment_name == Constants.PATCH_ASSESSMENT_SUMMARY and len(assessment_patch) > 0:
+                # Perform assessment truncation
+                assessment_patch, truncated_packages = self.__truncate_assessment_helper_func(assessment_patch, Constants.STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+
+                # self.__new_substatus_json_for_operation(Constants.PATCH_ASSESSMENT_SUMMARY, status, code, json.dumps(self.__assessment_summary_json))
+                # json.dumps(__assessment_summary_json) happens in multiple calls that creates escape \ for ", \ is an extra byte in the status file. it's unnecssary to perform json.dumps every time
+                # removing json.dumps will need to remove json.loads part in substatus_file_data["formattedMessage"]["message"] in entire codebase
+                # Reduce assement patch byte by quote size
+                quote_counts = sum(char == '"' for char in json.dumps(assessment_patch))
+
+                if (quote_counts + self.__get_byte_size(assessment_patch) > Constants.MAX_STATUS_FILE_SIZE_IN_BYTES):
+                    assessment_patch, truncated_packages = self.__truncate_assessment_helper_func(assessment_patch, Constants.STATUS_FILE_SIZE_LIMIT_IN_BYTES - quote_counts)
+
+                self.__truncated_patches.append(self.__set_truncated_package_detail("Assessment", truncated_packages))
+                self.composite_logger.log("Truncated assessment patches: ", self.__truncated_patches[0])
+
+            # Add assessment tombstone record
+            assessment_patch.append(self.__add_assessment_tombstone_record())
+
+            # Check for existing errors before recompose status file payload
+            code = self.__assessment_summary_json['errors']['code']
+            if code == 0 or code == 2:
```

Comment:
It's not easily readable  what code == 0 or code == 2 means. Use matching Constants for readability